auth.go

```diff
@@ -13,27 +16,54 @@ type AuthenticationResponse struct {
 	Scope        string `json:"scope"`
 }
 
-// Authenticate sends an authentication request, and returns a newly constructed client object
+var cache = ttlcache.New[string, string]()
+
+// Authenticate sends an authentication request when a token is not available in the cache, and returns a newly constructed client object
 func Authenticate(username, password, apiEndpoint string) (*Client, error) {
 	infolog.Printf("Start authentication into '%s' using '%s'", apiEndpoint, LoginUrl)
+	userAgent := ConstructUserAgentString()
+	cachedToken := getCachedAccessToken(username, apiEndpoint)
+	if len(cachedToken) > 0 {
```

Comment:
Nit: would it be clearer to return `nil` from `getCachedAccessToken` if not found? Than checking the length each time.

Reply:
I see, then possibly `String, error` as a return type would be a more go-like approach?

Reply:
But not having a token is acceptable and is not an error. Starting to use an error when a string is empty would mean having to treat errors differently when the type is a string. It's perfectly ok to return empty, and checking for emptiness is fine too (that's what they do in the [standard libs](http://golang.org/src/pkg/strconv/atoi.go))
Since we already return "" more than 20 times in the sdk, I think it is best to follow the same coding conventions